### PR TITLE
Feature/add color to patch files

### PR DESCRIPTION
**Pull request by mfehr**
This small fix adds the color information to the patch file, this way we don't have to parse the ply file too.

### CODE REVIEW

File: pmvs_catkin/src/pmvs/patchOrganizerS.cc [hunk 1]
```diff
@@ -96,7 +97,41 @@ void CpatchOrganizerS::writePatches2(const std::string prefix, bool bExportPLY,
     for (int p = 0; p < (int)m_ppatches.size(); ++p) {
       Cpatch patch = *m_ppatches[p];
       index2image(patch);
-      ofstr << patch << "\n";
+
+      Vec3i color;
+      int denom = 0;
+       Vec3f colorf;
+       for (int i = 0; i < (int)patch.m_images.size(); ++i) {
```
Comment (simonlynen) [inline pmvs_catkin/src/pmvs/patchOrganizerS.cc]: c++ style cast

File: pmvs_catkin/src/pmvs/patchOrganizerS.cc [hunk 2]
```diff
@@ -96,7 +97,41 @@ void CpatchOrganizerS::writePatches2(const std::string prefix, bool bExportPLY,
...
       index2image(patch);
-      ofstr << patch << "\n";
+
+      Vec3i color;
+      int denom = 0;
+       Vec3f colorf;
+       for (int i = 0; i < (int)patch.m_images.size(); ++i) {
+         const int image = patch.m_images[i];
+         colorf += m_fm.m_pss.getColor(patch.m_coord, image, m_fm.m_level);
+         denom++;
```
Comment (simonlynen) [inline pmvs_catkin/src/pmvs/patchOrganizerS.cc]: pre-increment

File: pmvs_catkin/src/pmvs/patchOrganizerS.cc [hunk 3]
```diff
@@ -96,7 +97,41 @@ void CpatchOrganizerS::writePatches2(const std::string prefix, bool bExportPLY,
     for (int p = 0; p < (int)m_ppatches.size(); ++p) {
       Cpatch patch = *m_ppatches[p];
       index2image(patch);
-      ofstr << patch << "\n";
+
+      Vec3i color;
+      int denom = 0;
```
Comment (simonlynen) [inline pmvs_catkin/src/pmvs/patchOrganizerS.cc:102]: variable name

File: pmvs_catkin/src/pmvs/patchOrganizerS.cc [hunk 4]
```diff
@@ -96,7 +97,41 @@ void CpatchOrganizerS::writePatches2(const std::string prefix, bool bExportPLY,
     for (int p = 0; p < (int)m_ppatches.size(); ++p) {
       Cpatch patch = *m_ppatches[p];
       index2image(patch);
-      ofstr << patch << "\n";
+
+      Vec3i color;
+      int denom = 0;
+       Vec3f colorf;
```
Comment (simonlynen) [inline pmvs_catkin/src/pmvs/patchOrganizerS.cc]: indentation

File: pmvs_catkin/src/pmvs/patchOrganizerS.cc [hunk 5]
```diff
@@ -96,7 +97,41 @@ void CpatchOrganizerS::writePatches2(const std::string prefix, bool bExportPLY,
...
+      Vec3i color;
+      int denom = 0;
+       Vec3f colorf;
+       for (int i = 0; i < (int)patch.m_images.size(); ++i) {
+         const int image = patch.m_images[i];
+         colorf += m_fm.m_pss.getColor(patch.m_coord, image, m_fm.m_level);
+         denom++;
+       }
+       colorf /= denom;
+       color[0] = min(255,(int)floor(colorf[0] + 0.5f));
```
Comment (simonlynen) [inline pmvs_catkin/src/pmvs/patchOrganizerS.cc]: c++ style casts, spaces after operators

File: pmvs_catkin/src/pmvs/patchOrganizerS.cc [hunk 6]
```diff
@@ -96,7 +97,41 @@ void CpatchOrganizerS::writePatches2(const std::string prefix, bool bExportPLY,
...
+
+      Vec3i color;
+      int denom = 0;
+       Vec3f colorf;
+       for (int i = 0; i < (int)patch.m_images.size(); ++i) {
+         const int image = patch.m_images[i];
+         colorf += m_fm.m_pss.getColor(patch.m_coord, image, m_fm.m_level);
+         denom++;
+       }
+       colorf /= denom;
```
Comment (simonlynen) [inline pmvs_catkin/src/pmvs/patchOrganizerS.cc]: CHECK_NE(denom, 0);

File: pmvs_catkin/src/pmvs/patchOrganizerS.cc [hunk 7]
```diff
@@ -96,7 +97,41 @@ void CpatchOrganizerS::writePatches2(const std::string prefix, bool bExportPLY,
...
+       ofstr << "PATCHS" << endl
+           << patch.m_coord << endl
+           << patch.m_normal << endl
+           << patch.m_ncc << ' '
+           << patch.m_dscale << ' '
+           << patch.m_ascale << endl
+           << color[0] << ' '
+           << color[1] << ' '
+           << color[2] << endl
+           << (int)patch.m_images.size() << endl;
```
Comment (simonlynen) [inline pmvs_catkin/src/pmvs/patchOrganizerS.cc]: c++ style cast

File: pmvs_catkin/src/pmvs/patchOrganizerS.cc [hunk 8]
```diff
@@ -96,7 +97,41 @@ void CpatchOrganizerS::writePatches2(const std::string prefix, bool bExportPLY,
...
+       for (int i = 0; i < (int)patch.m_images.size(); ++i)
```
Comment (simonlynen) [inline pmvs_catkin/src/pmvs/patchOrganizerS.cc]: c++ style cast

File: pmvs_catkin/src/pmvs/patchOrganizerS.cc [hunk 9]
```diff
@@ -96,7 +97,41 @@ void CpatchOrganizerS::writePatches2(const std::string prefix, bool bExportPLY,
...
+         ofstr << patch.m_images[i] << ' ';
+       ofstr << endl;
+
+       ofstr << (int)patch.m_vimages.size() << endl;
```
Comment (simonlynen) [inline pmvs_catkin/src/pmvs/patchOrganizerS.cc]: c++ style cast

File: pmvs_catkin/src/pmvs/patchOrganizerS.cc [hunk 10]
```diff
@@ -96,7 +97,41 @@ void CpatchOrganizerS::writePatches2(const std::string prefix, bool bExportPLY,
...
+       for (int i = 0; i < (int)patch.m_vimages.size(); ++i){
```
Comment (simonlynen) [inline pmvs_catkin/src/pmvs/patchOrganizerS.cc]: c++ style cast

File: pmvs_catkin/src/pmvs/patchOrganizerS.cc [hunk 11]
```diff
@@ -96,7 +97,41 @@ void CpatchOrganizerS::writePatches2(const std::string prefix, bool bExportPLY,
...
+         ofstr << patch.m_vimages[i] << ' ';
+       }
+       ofstr << endl;
+
+       ofstr << "\n";
```
Comment (simonlynen) [inline pmvs_catkin/src/pmvs/patchOrganizerS.cc]: use endl

File: pmvs_catkin/src/pmvs/patchOrganizerS.cc [hunk 12]
```diff
@@ -87,6 +87,7 @@ void CpatchOrganizerS::writePatches2(const std::string prefix, bool bExportPLY,
 
   if (bExportPatch)
   {
```
Comment (simonlynen) [inline pmvs_catkin/src/pmvs/patchOrganizerS.cc:89]: up

File: pmvs_catkin/src/pmvs/patchOrganizerS.cc [hunk 13]
```diff
@@ -87,6 +87,7 @@ void CpatchOrganizerS::writePatches2(const std::string prefix, bool bExportPLY,
 
   if (bExportPatch)
   {
+
```
Comment (simonlynen) [inline pmvs_catkin/src/pmvs/patchOrganizerS.cc:90]: vsp

**Comment by simonlynen** [comment]
I just saw that this is in pmvs. Feel free to ignore comments which are not in line with the pmvs style. Code style should be consistent with the rest of the package.

**Comment by simonlynen** [comment]
lgtm

**Comment by mfehr** [comment]
the code I added was taken from the ply file export, that's why it follows the pmvs code style.

**Pull request merged**